### PR TITLE
[Student][MBL-11282] Upload status view for file submissions

### PR DESCRIPTION
**Pull request by CalvinKern**
Click on file uploads for in-progress or failed is supported from the assignment details page. If you're watching an in-progress upload status, it can turn into the success state or the fail state. The only current ways I know to get a file error are file quota exceeded (submitting to a group assignment still has this problem) and submitting a file too large (we aren't preventing that yet).

### CODE REVIEW

File: apps/student/src/androidTest/java/com/instructure/student/ui/renderTests/UploadStatusSubmissionRenderTest.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Before
+    fun setup() {
+        baseModel = UploadStatusSubmissionModel(
+            submissionId = submissionId
+        )
+    }
+
+    @Test
+    fun displaysSuccessState() {
```
Comment (Blackchai) [inline apps/student/src/androidTest/java/com/instructure/student/ui/renderTests/UploadStatusSubmissionRenderTest.kt]: Don't forget to add some of those sick new metadata annotes!

Comment (CalvinKern) [reply apps/student/src/androidTest/java/com/instructure/student/ui/renderTests/UploadStatusSubmissionRenderTest.kt]: Oh snap!

File: apps/student/src/main/res/layout/fragment_upload_status_submission.xml [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        android:text="@string/retry"
+        android:textAllCaps="true"
+        android:textColor="@color/defaultActionColor"
+        android:textStyle="bold"
+        android:visibility="gone"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/uploadStatusStateButtonBarrier" />
+
+    <TextView
+        android:id="@+id/uploadStatusStateCancelAll"
```
Comment (Blackchai) [inline apps/student/src/main/res/layout/fragment_upload_status_submission.xml]: I feel like these three uploadStatusState textviews could benefit from sharing a style, but it is just within this layout. I'll let you decide.

File: apps/student/src/main/res/layout/viewholder_file_upload.xml [hunk 3]
```diff
@@ -51,15 +51,33 @@
...
+        app:layout_constraintStart_toStartOf="@id/fileName"
+        app:layout_constraintEnd_toEndOf="@id/fileName"
+        app:layout_constraintTop_toBottomOf="@id/fileSize"
+        tools:text="234 KB" />
+
+    <Space
+        android:layout_width="match_parent"
+        android:layout_height="16dp"
+        app:layout_constraintTop_toBottomOf="@id/fileError"
+        />
```
Comment (Blackchai) [inline apps/student/src/main/res/layout/viewholder_file_upload.xml]: eee!!!! A dangler!

File: apps/student/src/main/res/layout/fragment_upload_status_submission.xml (hunk 2 above)
Comment (CalvinKern) [reply apps/student/src/main/res/layout/fragment_upload_status_submission.xml]: I like it

**Pull request merged**